src/kakarot/kakarot_proxy.cairo

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts for Cairo v0.5.1 (upgrades/presets/Proxy.cairo)
+
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from starkware.starknet.common.syscalls import (
+    library_call,
+    library_call_l1_handler,
+    get_caller_address,
+)
+
+from openzeppelin.upgrades.library import Proxy
+from openzeppelin.access.accesscontrol.library import AccessControl
+
+const IMPLEMENTATION = 1641270636167208189312286704236493936886444818420034973493717770899220600387;  // implementation
```

Comment:
what are those?

Reply:
ahhh they're the int representation of keccak256("ADMIN") and keccak256("IMPLEMENTATION") as per OZ's access control's lib
maybe write it as comments:) + link to source if need be